PhysicsTools/Utilities/src/LumiReWeighting.cc

```diff
@@ -157,23 +157,7 @@ double LumiReWeighting::weight( float npv ) {
 
 double LumiReWeighting::weight( const edm::EventBase &e ) {
 
-  // find provenance of event objects, just to check at the job beginning if there might be an issue  
-
-  if(FirstWarning_) {
-
-    const edm::ProcessHistory& PHist = e.processHistory();
```

Comment:
NOTE: although the loop had no purpose, calling `e.processHistory()` does have a purpose. If there is a problem with the provenance this will throw an exception.